service/src/test/java/com/commercetools/pspadapter/tenant/TenantFactoryTest.java

```diff
@@ -0,0 +1,75 @@
+package com.commercetools.pspadapter.tenant;
+
+import com.commercetools.pspadapter.payone.config.PayoneConfig;
+import com.commercetools.pspadapter.payone.domain.ctp.CustomTypeBuilder;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TenantFactoryTest {
+
+    @Mock
+    private PayoneConfig payoneConfig;
+
+    @Mock
+    private TenantConfig tenantConfig;
+
+    private TenantFactory factory;
+
+    @Before
+    public void setUp() throws Exception {
+        when(payoneConfig.getApiUrl()).thenReturn("http://test.api.url");
+        when(tenantConfig.getPayoneConfig()).thenReturn(payoneConfig);
+        when(tenantConfig.getName()).thenReturn("testTenantName");
+
+        factory = new TenantFactory("testPayoneInterfaceName", tenantConfig);
+    }
+
+    @Test
+    public void getsTenantName() throws Exception {
+        assertThat(factory.getTenantName()).isEqualTo("testTenantName");
+    }
+
+    @Test
+    public void getsPayoneInterfaceName() throws Exception {
+        assertThat(factory.getPayoneInterfaceName()).isEqualTo("testPayoneInterfaceName");
+    }
+
+    @Test
+    public void getsPaymentHandlerUrl() throws Exception {
+        assertThat(factory.getPaymentHandlerUrl()).contains("/testTenantName/");
```

Comment:
but this unit test is exactly independent on base url, but Integration test - are dependent. This test _should not fail_ is someone changes the base URL, but integration tests - _should_.

This test case checks only the logic of URL building, but not the exact values.